ssl/statem/extensions_clnt.c

```diff
@@ -31,6 +31,28 @@ int tls_construct_ctos_renegotiate(SSL *s, WPACKET *pkt, unsigned int context,
     return 1;
 }
 
+int tls_construct_ctos_maxfragmentlen(SSL *s, WPACKET *pkt, unsigned int context,
+                                      X509 *x, size_t chainidx, int *al)
+{
+    /* Add Max Fragment Length extension if user enabled it. */
+    if (s->ext.max_fragment_length != TLSEXT_max_fragment_length_DISABLED) {
```

Comment:
Right...but is this correct behaviour? If the client does not explicitly specify a max_fragment_len in the CH for a resumed session, should it still be used? What happens if the client specifies one and it is different to the previously negotiated value - is that an error or does the previously negotiated or if not which one takes precedence? Are there any known implementations of this extension that we can do interop testing against?